pallets/amendments/src/lib.rs

```diff
@@ -21,88 +21,79 @@
+pub use pallet::*;
 
-    /// Origin that can veto amendments
-    type VetoOrigin: EnsureOrigin<Self::Origin>;
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
 
-    /// How much blocks have to be produced before executing the amendment
-    type Delay: Get<Self::BlockNumber>;
-}
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
-decl_error! {
-    pub enum Error for Module<T: Config> {
-        /// We failed to schedule the amendment
-        FailedToScheduleAmendment,
-        /// We failed to cancel the amendment
-        FailedToCancelAmendment,
-    }
-}
+        type Amendment: Parameter
+            + Dispatchable<Origin = Self::Origin>
+            + From<frame_system::Call<Self>>
+            + GetDispatchInfo;
 
-decl_event!(
-    pub enum Event<T>
-    where
-        <T as frame_system::Config>::BlockNumber,
-    {
-        /// A new amendment has been scheduled to be executed at the given block number
-        AmendmentScheduled(u64, BlockNumber),
-        /// An amendment has been vetoed and will never be triggered
-        AmendmentVetoed(u64),
-    }
-);
+        type Scheduler: ScheduleNamed<Self::BlockNumber, Self::Amendment, Self::PalletsOrigin>;
+        type PalletsOrigin: From<frame_system::RawOrigin<Self::AccountId>>;
 
-decl_storage! {
-    trait Store for Module<T: Config> as Amendments {
-        /// Internal variable to keep track of amendment ids for scheduling purposes
-        pub AmendmentsScheduled get(fn amendments_scheduled): u64;
+        /// Origin that can submit amendments
+        type SubmissionOrigin: EnsureOrigin<Self::Origin>;
+
+        /// Origin that can veto amendments
+        type VetoOrigin: EnsureOrigin<Self::Origin>;
+
+        /// How much blocks have to be produced before executing the amendment
+        type Delay: Get<Self::BlockNumber>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
     }
-}
 
-decl_module! {
-    /// The module declaration.
-    pub struct Module<T: Config> for enum Call where origin: T::Origin {
-        fn deposit_event() = default;
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
 
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
         /// Schedule `amendment` to be executed after the configured time, unless vetoed by `VetoOrigin`
-        #[weight = 100_000_000]
-        fn propose(origin, amendment: Box<T::Amendment>) -> DispatchResult {
+        #[pallet::weight(T::WeightInfo::propose(*_length_bound))]
+        pub fn propose(
+            origin: OriginFor<T>,
+            amendment: Box<T::Amendment>,
+            _length_bound: u32,
```

Comment:
I think we should check that `length_bound >= size_of_amendment` and throw an error if this is not the case

Reply:
Super Fast, Thanks for quick comment. Sure will rework on it :+1:

Reply:
Hello Eliott,

Completed refactoring weights annotation for `pub fn propose()`, as discussed removed the extra argument of ` _length_bound: u32`.